panic_handlers/console_panic/src/lib.rs

```diff
@@ -0,0 +1,21 @@
+#![no_std]
+use core::fmt::Write;
+use libtock_console::Console;
+use libtock_low_level_debug::{AlertCode, LowLevelDebug};
+use libtock_platform::{ErrorCode, Syscalls};
+use libtock_runtime::TockSyscalls;
+
+/// This handler requires some 0x400 bytes of stack
+
+#[panic_handler]
+fn panic_handler(info: &core::panic::PanicInfo) -> ! {
+    let mut writer = Console::<TockSyscalls>::writer();
+    if writeln!(writer, "{}", info).is_err() {
```

Comment:
As you have noticed, using the `Display` implementation on `core::panic::PanicInfo` is *extremely* expensive.

Instead of using `Display`, how about using `ufmt` (we have a fork of `ufmt` in the root of the repository) to print a subset of the panic info? You should be able to print `info.payload()` if it is a `&'static str` as well as `info.location()`.

Reply:
Hmm, switching to `ufmt` will collide with my in-progress work on adding an asynchronous API to the Console API.

This can probably wait until my asynchronous API is in place.